spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -354,10 +354,6 @@ private DistributionMode copyOnWriteMergeDistributionMode() {
     if (mergeModeName != null) {
       DistributionMode mergeMode = DistributionMode.fromName(mergeModeName);
       return adjustWriteDistributionMode(mergeMode);
-
-    } else if (table.spec().isPartitioned()) {
```

Comment:
This was actually done on purpose as inserts and merge operations have different use cases.